pkg/model/iam/tests/iam_builder_master_strict.json

```diff
@@ -0,0 +1,165 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeInstances",
+        "ec2:DescribeRouteTables",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVolumes"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sEC2MasterPermsDescribeResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateRoute",
+        "ec2:CreateSecurityGroup",
+        "ec2:CreateTags",
+        "ec2:CreateVolume",
+        "ec2:DeleteSecurityGroup",
+        "ec2:DeleteVolume",
+        "ec2:ModifyInstanceAttribute"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sEC2MasterPermsAllResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:AttachVolume",
+        "ec2:AuthorizeSecurityGroupIngress",
+        "ec2:DeleteRoute",
+        "ec2:DetachVolume",
+        "ec2:RevokeSecurityGroupIngress"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Condition": {
+        "StringEquals": {
+          "ec2:ResourceTag/KubernetesCluster": "iam-builder-test.k8s.local"
+        }
+      },
+      "Sid": "kopsK8sEC2MasterPermsTaggedResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "autoscaling:DescribeAutoScalingGroups",
+        "autoscaling:DescribeLaunchConfigurations",
+        "autoscaling:GetAsgForInstance"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sASMasterPermsAllResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "autoscaling:SetDesiredCapacity",
+        "autoscaling:TerminateInstanceInAutoScalingGroup",
+        "autoscaling:UpdateAutoScalingGroup"
+      ],
+      "Resource": [
+        "*"
```

Comment:
AutoScaling now has resource level permissions so instead of tags you also have the option of using a prefix on your AutoScalingGroup and LaunchConfiguration names.

Reply:
I like that idea

Reply:
That would be good to add in. Would you be able to point me in the right direction of retrieving the autoscaling & launch configuration names and account id to populate the resources list?